mp/src/game/client/momentum/ui/HUD/hud_mapfinished.cpp

```diff
@@ -425,25 +443,85 @@ void CHudMapFinishedDialog::SetCurrentPage(int pageNum)
 
     m_pZoneSync2->SetText(CConstructLocalizedString(m_iCurrentPage == 0 ? m_pwSync2Overall : m_pwSync2Zone,
-                                                    m_pRunStats ? m_pRunStats->GetZoneStrafeSync2Avg(m_iCurrentPage) : 0.0f));
+                                                    m_pRunStats->GetZoneStrafeSync2Avg(m_iCurrentPage)));
 
     int velType = m_cvarVelType.GetInt();
-    m_pZoneVelEnter->SetText(CConstructLocalizedString(m_pwVelZoneEnter, m_pRunStats ? m_pRunStats->GetZoneEnterSpeed(m_iCurrentPage, velType) : 0.0f));
+    m_pZoneVelEnter->SetText(CConstructLocalizedString(m_pwVelZoneEnter, m_pRunStats->GetZoneEnterSpeed(m_iCurrentPage, velType)));
+
+    m_pZoneVelExit->SetText(CConstructLocalizedString(m_pwVelZoneExit, m_pRunStats->GetZoneExitSpeed(m_iCurrentPage, velType)));
+
+    m_pZoneVelAvg->SetText(CConstructLocalizedString(m_pwVelAvg, m_pRunStats->GetZoneVelocityAvg(m_iCurrentPage, velType)));
+
+    m_pZoneVelMax->SetText(CConstructLocalizedString(m_pwVelMax, m_pRunStats->GetZoneVelocityMax(m_iCurrentPage, velType)));
+}
+
+void CHudMapFinishedDialog::LoadPlayerBestTime()
+{
+    const auto pPlayer = C_MomentumPlayer::GetLocalMomPlayer();
+    if (!pPlayer)
+        return;
+
+    const auto pRunData = pPlayer->GetCurrentUIEntData();
+
+    CMomReplayBase *pBase = MomUtil::GetBestTime(g_pGameRules->MapName(), pRunData->m_flTickRate,
+                                                 pRunData->m_iCurrentTrack, pRunData->m_iRunFlags);
+
+    if (!pBase)
+        return;
+    
+    m_pPbRunStats.FullyCopyFrom(*pBase->GetRunStats());
+
+    delete pBase;
+}
+
+void CHudMapFinishedDialog::GetDiffString(float diff, char* compareStringOut, Color* compareColorOut) 
+{
+    char tempANSITimeOutput[BUFSIZETIME];
+
+    if (CloseEnough(diff, 0.0f, FLT_EPSILON))
+    {
+        compareColorOut->SetRawColor(m_cTie.GetRawColor());
+        MomUtil::FormatTime(diff, tempANSITimeOutput);
+        V_snprintf(compareStringOut, BUFSIZELOCL, "(%s)", tempANSITimeOutput);
+        return;
+    }
+
+    char diffSign = diff > 0.0f ? '+' : '-';
+    compareColorOut->SetRawColor(diff > 0.0f ? m_cLoss.GetRawColor() : m_cGain.GetRawColor());
+
+    diff = fabs(diff);
+    MomUtil::FormatTime(diff, tempANSITimeOutput);
+    V_snprintf(compareStringOut, BUFSIZELOCL, "(%c %s)", diffSign, tempANSITimeOutput);
+}
+
+void CHudMapFinishedDialog::SetComparisonForPage(int iPage)
+{
+
+    if (m_pPbRunStats.GetZoneTicks(0) == 0)
+    {
+        m_pComparisonLabel->SetText("");
+        return;
+    }
+
+    int runTicks = iPage == 0 ? m_pRunData->m_iRunTime : m_pRunStats->GetZoneTicks(iPage);
 
-    m_pZoneVelExit->SetText(CConstructLocalizedString(m_pwVelZoneExit, m_pRunStats ? m_pRunStats->GetZoneExitSpeed(m_iCurrentPage, velType) : 0.0f));
+    float diff = (static_cast<float>(runTicks) - m_pPbRunStats.GetZoneTicks(iPage)) * m_pRunData->m_flTickRate;
```

Comment:
`int`s can be subtracted first then casted into a float:
```suggestion
    float diff = float(runTicks - m_pPbRunStats.GetZoneTicks(iPage)) * m_pRunData->m_flTickRate;
```